### PR TITLE
[23.0] Introduce `--disable-debuginfo-stripping` option

### PR DESCRIPTION
**Pull request by zakkak**
Enables the use of `--disable-debuginfo-stripping` which was introduced
in https://github.com/oracle/graal/pull/7624

### CODE REVIEW

**Comment by zakkak** [comment]
Follow up PRs will be created for `23.1`, `24.0`, and `master` once this gets approved.

**Review by jerboaa — COMMENTED** [review]
Patch itself is fine, but could you update the workflow file to check out the right `mx` version, like our mandrel CI does?

PR checker fails due to:

```
2024-01-11T13:21:02.9038428Z Exception in thread "main" java.lang.RuntimeException: java.nio.file.NoSuchFileException: /home/runner/work/mandrel-packaging/mandrel-packaging/mandrel/substratevm/mxbuild/linux-amd64/com.oracle.svm.native.libchelper/amd64/liblibchelper.a
2024-01-11T13:21:02.9038637Z 	at FileSystem.copy(build.java:1529)
2024-01-11T13:21:02.9038742Z 	at build.main(build.java:182)
2024-01-11T13:21:02.9039805Z Caused by: java.nio.file.NoSuchFileException: /home/runner/work/mandrel-packaging/mandrel-packaging/mandrel/substratevm/mxbuild/linux-amd64/com.oracle.svm.native.libchelper/amd64/liblibchelper.a
```

Suggesting it's using mx suitable for master, only.

**Comment by zakkak** [comment]
Thanks for mentioning this, I was struggling to see what's wrong... :)

File: .github/workflows/buildJDK.yml [hunk 1]
```diff
@@ -45,12 +45,11 @@ jobs:
...
-      with:
-        repository: graalvm/mx.git
-        fetch-depth: 1
-        ref: master
-        path: ${{ github.workspace }}/mx
+    - name: Checkout MX
+      run: |
+          VERSION=$(jq -r .mx_version ${MANDREL_REPO}/common.json)
+          git clone ${GITHUB_SERVER_URL}/graalvm/mx --depth 1 --branch ${VERSION} ${MX_HOME}
+          ./mx/mx --version    - uses: actions/checkout@v3
```
Comment (jerboaa) [inline .github/workflows/buildJDK.yml:52]: ` - uses actions/checkout@v3` intentional?

File: .github/workflows/buildJDK.yml [hunk 2]
```diff
@@ -135,12 +134,11 @@ jobs:
...
```
Comment (jerboaa) [inline .github/workflows/buildJDK.yml:141]: same here.

File: .github/workflows/buildJDK.yml [hunk 3]
```diff
@@ -242,12 +240,11 @@ jobs:
...
```
Comment (jerboaa) [inline .github/workflows/buildJDK.yml:247]: and here.

File: .github/workflows/buildJDK.yml (hunk 1 above)
Comment (zakkak) [reply .github/workflows/buildJDK.yml:52]: :facepalm: , no, thanks for the catch!

**Comment by jerboaa** [comment]
@zakkak Oh, I've just realized this doesn't yet work on Windows:

```
 VERSION=$(jq -r .mx_version ${MANDREL_REPO}/common.json): D:\a\_temp\9b0ef8ba-4f46-4891-af53-35e0c27ce28b.ps1:2
Line |
   2 |  VERSION=$(jq -r .mx_version ${MANDREL_REPO}/common.json)
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The term 'VERSION=$(jq -r .mx_version ${MANDREL_REPO}/common.json)' is not recognized as a name of a cmdlet,
     | function, script file, or executable program. Check the spelling of the name, or if a path was included, verify
     | that the path is correct and try again.
```

File: .github/workflows/buildJDK.yml [hunk 4]
```diff
@@ -135,12 +134,11 @@ jobs:
...
-      with:
-        repository: graalvm/mx.git
-        fetch-depth: 1
-        ref: master
-        path: ${{ github.workspace }}/mx
+    - name: Checkout MX
+      run: |
+          VERSION=$(jq -r .mx_version ${MANDREL_REPO}/common.json)
+          git clone ${GITHUB_SERVER_URL}/graalvm/mx --depth 1 --branch ${VERSION} ${MX_HOME}
+          ./mx/mx --version
```
Comment (jerboaa) [inline .github/workflows/buildJDK.yml:141]: This should probably be using bash:

```
      shell: bash
```

... is what's used in the mandrel CI.

**Review by jerboaa — CHANGES_REQUESTED** [review]

**Review by jerboaa — APPROVED** [review]
Nice. GHA clean now.

**Pull request merged**